BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/IndoorPositioning.java

```diff
@@ -97,7 +99,20 @@ private void updateLocation() {
         return usableBeacons;
     }
 
+    public static Location walkingSpeedFilter(Location oldLocation, Location newLocation) {
+        double distance = oldLocation.getDistanceTo(newLocation);
+        double angle = oldLocation.getAngleTo(newLocation);
+        if (distance > HUMAN_WALKING_SPEED) {
+            return oldLocation.getShiftedLocation(HUMAN_WALKING_SPEED, angle);
```

Comment:
You assume that the timestamp delta between any two given locations is one second, which might not be the case